### ISSUE TITLE
Support s/defprotocol from prismatic schema

### ISSUE THREAD

**Issue by lucasmafra** [clj-kondo/clj-kondo#2350]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

`2024.05.24`

**platform** 

JVM

**problem**

Version [1.4.0 of prismatic schema](https://github.com/plumatic/schema/blob/master/CHANGELOG.md) introduced a new macro `s/defprotocol`. However, clj-kondo currently does not support that macro.

**repro**
```clj
(ns example
  (:require
   [schema.core :as s]))

(s/defprotocol MyProtocol
  (my-method :- s/Any
    [this :- s/Any]))

(my-method {})
```

```sh
error: [invalid-arity] example/my-method is called with 1 arg but expects 3
```

**expected behavior**

Support `s/defprotocol`

[ Optional: if you are interested in doing a PR yourself, please leave a note. ]
I can do the fix and link the PR here

**Comment by emerson-matos** [comment]
is this still a problem right? can I work on this?

**Comment by emerson-matos** [comment]
@borkdude first time working with clj-kondo codebase, let me know if you have any tips in mind on how to approach this

**Comment by borkdude** [comment]
@emerson-matos Hey, thanks for helping out. It should be relatively straightforward when you look at the other examples like `s/defn`. What happens is basically stripping the type annotations and then linting it like the normal Clojure equivalent.

**Comment by emerson-matos** [comment]
@borkdude sorry about the tagging, but I think it is done, let me know if you want me to change/adapt anything :)

**Comment by borkdude** [comment]
@emerson-matos No worries! I'll take a look tomorrow hopefully!
